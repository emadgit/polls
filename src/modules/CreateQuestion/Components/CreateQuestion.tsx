import React, { FunctionComponent } from "react";
import { WrappedProps } from "../Containers/CreateComponent";
import { Form } from "formik";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import Grid from "@material-ui/core/Grid";
import Snackbar from "@material-ui/core/Snackbar";
import SnackbarContent from "@material-ui/core/SnackbarContent";

export const CreateQuestion: FunctionComponent<WrappedProps> = ({
  isSubmitting,
  handleChange,
  values,
  history,
  errors
}) => {
  return (
    <div>
      <Form>
        <Grid container spacing={24} style={{ padding: 20 }}>
          <Grid item xs={12} md={6} lg={6}>
            <TextField
              id="question"
              name="question"
              label="Question"
              onChange={handleChange}
              margin="normal"
              InputLabelProps={{
                style: { color: "#fff" }
              }}
              InputProps={{
                style: { color: "#00cea5" }
              }}
              fullWidth={true}
              value={values.question}
            />
            {errors.question && (
              <p style={{ color: "red" }}>{errors.question}</p>
            )}
          </Grid>

          <Grid item xs={12} md={6} lg={6}>
            <TextField
              id="question"
              name="choices"
              label="Choices"
              margin="normal"
              fullWidth={true}
              InputLabelProps={{
                style: { color: "#fff" }
              }}
              InputProps={{
                style: { color: "#00cea5" }
              }}
              style={{ color: "#fff" }}
              onChange={handleChange}
              value={values.choices}
            />
            {errors.choices && <p style={{ color: "red" }}>{errors.choices}</p>}
          </Grid>

          <Grid item xs={12} md={6} lg={6}>
            <Button
              variant="contained"
              color="primary"
              type="submit"
              disabled={isSubmitting}
            >
              Create Question
            </Button>
          </Grid>
        </Grid>
      </Form>

      <Snackbar
        anchorOrigin={{
          horizontal: "right",
          vertical: "bottom"
        }}
        autoHideDuration={2000}
        open={isSubmitting}
        onClose={() => history.goBack()}
        ContentProps={{
          "aria-describedby": "message-id"
        }}
      >
        <SnackbarContent
          style={{ backgroundColor: "green" }}
          message={<span id="message-id">Your question created!</span>}
        />
      </Snackbar>
    </div>
  );
};
