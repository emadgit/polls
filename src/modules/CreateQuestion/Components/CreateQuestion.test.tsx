import * as React from "react";
import { shallow } from "enzyme";
import { CreateQuestion } from "./CreateQuestion";

let mock: any = jest.fn();

const mockData = {
  values: {
    question: "",
    choices: ""
  },
  errors: {
    question: "",
    choices: ""
  }
};

// @ts-ignore
const component = shallow(<CreateQuestion {...mockData} />);

describe("CreateQuestion Component", () => {
  it("should render correctly", () => {
    expect(component).toMatchSnapshot();
  });
});
