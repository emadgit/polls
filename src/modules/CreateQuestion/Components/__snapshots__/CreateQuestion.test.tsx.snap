// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateQuestion Component should render correctly 1`] = `
<div>
  <Form>
    <WithStyles(Grid)
      container={true}
      spacing={24}
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <WithStyles(Grid)
        item={true}
        lg={6}
        md={6}
        xs={12}
      >
        <TextField
          InputLabelProps={
            Object {
              "style": Object {
                "color": "#fff",
              },
            }
          }
          InputProps={
            Object {
              "style": Object {
                "color": "#00cea5",
              },
            }
          }
          fullWidth={true}
          id="question"
          label="Question"
          margin="normal"
          name="question"
          required={false}
          select={false}
          value=""
          variant="standard"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        lg={6}
        md={6}
        xs={12}
      >
        <TextField
          InputLabelProps={
            Object {
              "style": Object {
                "color": "#fff",
              },
            }
          }
          InputProps={
            Object {
              "style": Object {
                "color": "#00cea5",
              },
            }
          }
          fullWidth={true}
          id="question"
          label="Choices"
          margin="normal"
          name="choices"
          required={false}
          select={false}
          style={
            Object {
              "color": "#fff",
            }
          }
          value=""
          variant="standard"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        lg={6}
        md={6}
        xs={12}
      >
        <WithStyles(Button)
          color="primary"
          type="submit"
          variant="contained"
        >
          Create Question
        </WithStyles(Button)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </Form>
  <WithStyles(Snackbar)
    ContentProps={
      Object {
        "aria-describedby": "message-id",
      }
    }
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "bottom",
      }
    }
    autoHideDuration={2000}
    onClose={[Function]}
  >
    <WithStyles(SnackbarContent)
      message={
        <span
          id="message-id"
        >
          Your question created!
        </span>
      }
      style={
        Object {
          "backgroundColor": "green",
        }
      }
    />
  </WithStyles(Snackbar)>
</div>
`;
